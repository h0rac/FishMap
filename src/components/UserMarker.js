import React, { Component } from 'react';
import MapView from 'react-native-maps';

class UserMarker extends Component {


	render() {

		const region = {
			latitude: this.props.marker.latitude,
			longitude: this.props.marker.longitude

		};
		return (
			<MapView.Marker
				coordinate={region}
			>
			</MapView.Marker>
		);
	}
}

export default UserMarker;



