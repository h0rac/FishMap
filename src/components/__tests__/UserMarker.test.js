import 'react-native';
import React from 'react';
import UserMarker from '../UserMarker';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('Testing UserMarker component', () => {

	let tree;
	beforeEach(() => {
        const marker =  {
            latitude: 54.475408,
            longitude: 18.263086
        }
        tree = shallow(
		<UserMarker marker={marker}/>);
	});
	it('renders correctly', () => {
		expect(tree).toMatchSnapshot();
	});
})