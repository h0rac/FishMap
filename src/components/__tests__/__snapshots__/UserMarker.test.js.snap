// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserMarker component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserMarker
    marker={
      Object {
        "latitude": 54.475408,
        "longitude": 18.263086,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "coordinate": Object {
        "latitude": 54.475408,
        "longitude": 18.263086,
      },
      "onPress": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "coordinate": Object {
          "latitude": 54.475408,
          "longitude": 18.263086,
        },
        "onPress": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
